ClinGen/clincoded
# Connects to #1010. Add warning message in disease modal.

**Known issues:**
1. At this point, deleting the associated disease only invoke the provisional checkbox to be unchecked if it was previously marked as provisional, assuming the calculated/modified pathogenicity are either **Likely Pathogenic** and/or **Pathogenic**.
**(Update as of 9-28: with the latest commit, I am now able to uncheck the provisional checkbox (and disable it) in the UI and reset the provisional to "In Progress" in the database simultaneously, in the event in which the user deletes the associated disease from the interpretation while the calculated/modified pathogenicity are either Likely Pathogenic or Pathogenic.)**
2. In the disease modal component, the display of the warning message is only using one condition - the `markAsProvisional` property in the `interpretation` object. We are not yet using the stored calculated pathogenicity nor the stored/in-app modified pathogenicity values as additional conditions.

**Technical details:**
1. The **markAsProvisional** property was added to the array used for the flatten interpretation object in the **curator.js** so that saving a change in associated disease would not wipe out the pre-existing **markAsProvisional** field in the interpretation object as a result of saving the provisional form.

**Steps to test:**
1. Login and use 15333 variant_id at the variation selection interface.
2. Start a new interpretation and evaluate various criteria so that the calculated pathogenicity is either **Likely Pathogenic** or **Pathogenic**.
3. Associate the interpretation with a disease and proceed to the Summary page.
4. Check the provisional interpretation checkbox, without making a **Modify Pathogenicity** selection, and Save the form.
5. Return to the interpretation and edit the disease. Remove the disease in the modal and expect to see the yellow warning message to appear.
6. Save the disease removal change and proceed to the Summary page again. Expect to see the previously marked provisional checkbox is now unchecked.
